modules/user-data/templates/06_application_insights_config.sh.tpl

```diff
@@ -6,12 +6,93 @@ echo "###########################################"
 echo "#    Configuring Application Insights     #"
 echo "###########################################"
 
+RECORD_NAME=$(cat /tmp/node_name)
 
 # Overrides the config file
 cat <<-EOF > /opt/graphdb/applicationinsights.json
 {
-  "connectionString": "${appi_connection_string}"
+    "role": {
+        "name": "$RECORD_NAME"
+      },
+    "connectionString": "${appi_connection_string}",
+    "sampling": {
+      "percentage": ${appi_sampling_percentage}
+  },
+    "instrumentation": {
+        "logging": {
+            "level": "${appi_logging_level}"
+        }
+    },
+    "preview": {
+        "captureLogbackMarker": true,
+        "captureLog4jMarker": true,
+        "captureLogbackCodeAttributes": true,
+        "sampling": {
+            "overrides": [
+              {
+                "telemetryType": "dependency",
+                "percentage": ${appi_dependency_sampling_override}
+              },
+              {
+                "telemetryType": "request",
+                "percentage": 0,
+                "attributes": [
+                   {
+                     "key": "rpc.service",
+                     "value": ".*cluster.*",
+                     "matchType":  "regexp"
+                   }
+                ]
+              },
+              {
+                "telemetryType": "request",
+                "percentage": 0,
+                "attributes": [
+                   {
+                     "key": "http.url",
+                     "value": "https?://[^/]+/(rest|monitor|res|i18n|js|css|img|pages)/.*",
```

Comment:
Filtering by content type should be a better approach but this is simpler, I guess.